gtsfm/frontend/verifier/gric_verifier.py

```diff
@@ -0,0 +1,191 @@
+                instead of fundamental matrix. This should be preferred when the exact intrinsics are known.
+            estimation_threshold_px: maximum distance (in pixels) to consider a match an inlier, under squared
+                Sampson distance.
+        """
+        self._use_intrinsics_in_verification = use_intrinsics_in_verification
+        self._estimation_threshold_px = estimation_threshold_px
+        self._min_matches = (
+            verifier_base.NUM_MATCHES_REQ_E_MATRIX
+            if self._use_intrinsics_in_verification
+            else verifier_base.NUM_MATCHES_REQ_F_MATRIX
+        )
+
+        # for failure, i2Ri1 = None, and i2Ui1 = None, and no verified correspondences, and inlier_ratio_est_model = 0
+        self._failure_result = (None, None, np.array([], dtype=np.uint64), 0.0)
+
+    def __estimate_two_view_geometry(
+        self,
+        uv_i1: np.ndarray,
+        uv_i2: np.ndarray,
+        camera_intrinsics_i1: Cal3Bundler,
+        camera_intrinsics_i2: Cal3Bundler,
+    ) -> Dict[str, Any]:
+        """Use the pycolmap Pybind wrapper to estimate an Essential matrix using LORANSAC.
+
+        Args:
+            uv_i1: array of shape (N3,2) representing coordinates of 2d points in image 1.
+            uv_i2: array of shape (N3,2) representing corresponding coordinates of 2d points in image 2.
+            camera_intrinsics_i1: intrinsics for image i1.
+            camera_intrinsics_i2: intrinsics for image i2.
+
+        Returns:
+            dictionary containing result status code, estimated relative pose (R,t), and inlier mask.
+        """
+
+        def get_pycolmap_camera(camera_intrinsics: Cal3Bundler) -> pycolmap.Camera:
+            """Convert Cal3Bundler intrinsics to a pycolmap-compatible format (a dictionary).
+
+            See https://colmap.github.io/cameras.html#camera-models for info about the COLMAP camera models.
+            Both SIMPLE_PINHOLE and SIMPLE_RADIAL use 1 focal length.
+            """
+            focal_length = camera_intrinsics.fx()
+            cx, cy = camera_intrinsics.px(), camera_intrinsics.py()
+
+            # TODO (johnwlambert): use more accurate proxy?
+            width = int(cx * 2)
+            height = int(cy * 2)
+            logger.info("Estimated height and width: %d,%d", height, width)
+
+            camera_dict = pycolmap.Camera(
+                model="SIMPLE_PINHOLE",
+                width=width,
+                height=height,
+                params=[focal_length, cx, cy],
+            )
+            return camera_dict
+
+        camera_dict1 = get_pycolmap_camera(camera_intrinsics_i1)
+        camera_dict2 = get_pycolmap_camera(camera_intrinsics_i2)
+
+        result_dict = pycolmap.two_view_geometry_estimation(
+            points2D1=uv_i1,
+            points2D2=uv_i2,
+            camera1=camera_dict1,
+            camera2=camera_dict2,
```

Comment:
just call the `get_pycolmap_camera` here instead of assigning to local variables? I feel `camera_dict` is not very clear on what it is, and we can just avoid the confusion.

Reply:
Good idea, thanks. Updated now.